### DISCUSSION TITLE
Joining 3 tables with nested relationship and counting

### DISCUSSION THREAD

**Discussion by ericmartinezr** [prisma/prisma#7426] [Q&A]
Hey guys! This is my first question here and I'm new to Prisma.

I want to join 3 tables, where 2 of them have no direct relationship and get the count of the result. The query manually written is super simple but I can't figure it out how to write it with Prisma.

The relationship is like this

`Proyecto > EstadoChecklist > Incidencia`

```js
model Proyecto {
  codigo        Int          @id
  nombre        String       @db.VarChar(100)
  estadoChecklist EstadoChecklist?
  @@map("proyecto")
}

model EstadoChecklist {
  id                  Int     @id @default(autoincrement())
  codigo              Int
  proyecto   Proyecto?   @relation(fields: [codigo], references: [codigo])
  incidencia Incidencia?
  @@map("estado_checklist")
}

model Incidencia {
  id_estado_checklist Int     @id
  estado              Boolean
  estadoChecklist EstadoChecklist?       @relation(fields: [id_estado_checklist], references: [id])
  @@map("incidencia")
}
```

My Prisma query is as follows:

```js
this.prisma.proyecto
  .groupBy({
        by: ['codigo', 'nombre'],
        where: {
          estadoChecklist: {
            incidencia: {
              estado: {
                equals: true
              }
            }
          }
        },
        _count: {
          codigo: true
        }
      })
```

Which returns as follows (almost what I want, but the count is not right)

```
[
  { codigo: 485, nombre: '0485_BCI_Asdzc', _count: { codigo: 1 } },
  { codigo: 490, nombre: '0490_BCI_Asd', _count: { codigo: 1 } }
]
```

And produces the following query
```sql
SELECT `proyecto`.`codigo`, `proyecto`.`nombre`, COUNT(*) 
FROM `proyecto` WHERE (`proyecto`.`codigo`) 
IN (
	SELECT `t0`.`codigo` FROM `proyecto` AS `t0` 
	INNER JOIN `estado_checklist` AS `j0` ON (`j0`.`codigo`) = (`t0`.`codigo`) 
	WHERE ((`j0`.`id`) IN (SELECT `t1`.`id` FROM `estado_checklist` AS `t1`
	INNER JOIN `incidencia` AS `j1` ON (`j1`.`id_estado_checklist`) = (`t1`.`id`) 
	WHERE (`j1`.`estado` = 1 AND `t1`.`id` IS NOT NULL)) AND `t0`.`codigo` IS NOT NULL)
) 
GROUP BY `proyecto`.`codigo`, `proyecto`.`nombre`
```
The inner select is fine and it returns the right amount of rows (7 and 1), but the outer select is not correct and I don't know how to continue (I can workaround by writing a raw query for now).

If I write the query manually as follows it works just fine
```sql
SELECT p.codigo, p.nombre, COUNT(1) as total_incidencias FROM proyecto p
     INNER JOIN estado_checklist ec 
     ON p.codigo = ec.codigo
     INNER JOIN incidencia i
     ON ec.id = i.id_estado_checklist AND i.estado = 1
     GROUP BY p.codigo, p.nombre
```
Result
```
[
  { codigo: 485, nombre: '0485_BCI_Asdzc', total_incidencias: 7 },
  { codigo: 490, nombre: '0490_BCI_Asd', total_incidencias: 1 }
]
```

If you need more info let me know.

Thanks in advance.

**Comment by ryands17** [comment]
@ericmartinezr 👋 

Could you replace `codigo` with `_all` In count and see if it works? If it doesn't, it would be great if you could create a feature request [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) for having nested relations count in group by.

**Reply by ericmartinezr** [reply]
Thank you for your answer @ryands17 . I tried with `_all: true` and I got the same result. I will open the feature request as you suggested. Hopefully they will accept it, or if I made a mistake they'll let me know how to do it correctly :P.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
